src/coreclr/jit/lclvars.cpp

```diff
@@ -882,6 +882,15 @@ void Compiler::lvaInitUserArgs(InitVarDscInfo* varDscInfo, unsigned skipArgs, un
                 argRegTypeInStruct1 = (floatFlags & STRUCT_FIRST_FIELD_SIZE_IS8) ? TYP_DOUBLE : TYP_FLOAT;
                 argRegTypeInStruct2 = (floatFlags & STRUCT_SECOND_FIELD_SIZE_IS8) ? TYP_LONG : TYP_INT;
             }
+#ifdef TARGET_RISCV64
+            else if ((floatFlags | STRUCT_FLOAT_FIELD_SECOND | STRUCT_FIRST_FIELD_SIZE_IS8) == floatFlags)
+            {
+                canPassArgInRegisters = varDscInfo->canEnreg(TYP_LONG, 2);
+
+                argRegTypeInStruct1 = TYP_LONG;
+                argRegTypeInStruct2 = TYP_LONG;
+            }
+#endif // TARGET_RISCV64
```

Comment:
This seems like it should be fixed in `MethodTable::GetRiscv64PassStructInRegisterFlags` instead. If these should not be passed in float registers then that function should not be returning that we should.

Reply:
Thank you. I will fix it.